x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.js

```diff
@@ -56,7 +57,19 @@ export class HeadlessChromiumDriverFactory {
 
       const driver$ = message$
         .first(line => line.indexOf(`DevTools listening on ws://127.0.0.1:${bridgePort}`) >= 0)
-        .do(() => this.logger.debug('Connecting chrome remote inspector'))
+        .do(() => {
+          // See https://github.com/elastic/kibana/issues/19351 for why this is necessary. Long story short, on certain
+          // linux platforms (fwiw, we have only experienced this on jenkins agents) the first bootup of chromium takes
+          // a long time doing something with fontconfig packages loading up a cache. The cdp command will timeout
+          // if we don't wait for this manually. Note that this may still timeout based on the value of
+          // xpack.reporting.queue.timeout. Subsequent runs should be fast because the cache will already be
+          // initialized.
+          this.logger.debug('Ensure chromium is running and listening');
+          spawnSync(
+            `curl`,
```

Comment:
I think this approach is fine; however, I'd recommend against using `spawnSync `with `curl` to see if chromium's dev tools protocol is listening/active. Not all servers will have `curl` and `spawnSync` will lock the event loop while we wait for it to respond. I'd recommend using [http.request](https://nodejs.org/api/http.html#http_http_request_options_callback) asynchronously to perform this logic.